### PR TITLE
Bug fix, pass report data if healthcheck failed with Exception

### PR DESCRIPTION
**Pull request by AndreyTretyak**
Fix for a bug found by @artempushkin, currently, report data ignored in case of exception.

### CODE REVIEW

**Review by vlad-ion — APPROVED** [review]

**Review by artempushkin — APPROVED** [review]

File: src/Extensions/Diagnostics.HealthChecks/HealthCheckParameters.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+			{
+				ReportData = s_emptyDictionary;
+			}
+			else
+			{
+				// should be replaced by passing enumerable to constructor after we drop full framework support
+				Dictionary<string, object> dictionary = new Dictionary<string, object>(reportData.Length);
+				foreach (KeyValuePair<string, object> pair in reportData)
+				{
+					dictionary.Add(pair.Key, pair.Value);
```
Comment (K-Cully) [inline src/Extensions/Diagnostics.HealthChecks/HealthCheckParameters.cs:34]: This assumes every Key in the array of KeyValuePair objects is unique and could throw.

**Review by K-Cully — CHANGES_REQUESTED** [review]

File: src/Extensions/Diagnostics.HealthChecks/HealthCheckParameters.cs (hunk 1 above)
Comment (K-Cully) [reply src/Extensions/Diagnostics.HealthChecks/HealthCheckParameters.cs:34]: When fixing this, please also add UT coverage for this case.

Comment (AndreyTretyak) [reply src/Extensions/Diagnostics.HealthChecks/HealthCheckParameters.cs:34]: What would be the proper behavior for you here?
Do you think we should override values with the same key?

Comment (AndreyTretyak) [reply src/Extensions/Diagnostics.HealthChecks/HealthCheckParameters.cs:34]: Discussed in a call we'll add a more descriptive error message here.

Comment (AndreyTretyak) [reply src/Extensions/Diagnostics.HealthChecks/HealthCheckParameters.cs:34]: The discussion moved here: https://github.com/microsoft/Omex/pull/237

**Pull request merged**